src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -404,12 +404,12 @@ private static bool FindInGac(string gacRoot, AssemblyName assemblyName, out str
             if (Directory.Exists(tempAssemblyDirPath))
             {
                 // Enumerate all directories, sort by name and select the last. This selects the latest version.
-                var chosenVersionDirectory = Directory.GetDirectories(tempAssemblyDirPath).OrderBy(d => d).LastOrDefault();
+                var chosenVersionDirectory = Directory.EnumerateDirectories(tempAssemblyDirPath).OrderBy(d => d).LastOrDefault();
```

Comment:
Wouldn't it then be better to do an `.OrderByDescending(d=>d).FirstOrDefault()` ?

Reply:
> Wouldn't it then be better to do an `.OrderByDescending(d=>d).FirstOrDefault()` ?

`EnumerateDirectories` generally returns elements in ascending order, although this is definitely not guaranteed, and is dependent on the file system. So I'm not sure whether there would be any benefit to sorting by descending order. In any case such changes would be outside of the scope of this PR.